File: src/test_model_embedding.c
// test_model_embedding.c - Test program for model embedding interface
#include <stdio.h>
#include <stdlib.h>
#include "../include/c_model_interface.h"

int main() {
    printf("Testing model embedding interface...\n");
    
    // Test getting embedded model
    const uint8_t* model_data = c_get_embedded_model();
    if (model_data == NULL) {
        printf("Failed to get embedded model data\n");
        return 1;
    }
    
    size_t model_size = c_get_embedded_model_size();
    printf("Got embedded model of size: %zu bytes\n", model_size);
    
    // Test validating embedded model
    int is_valid = c_validate_embedded_model();
    printf("Embedded model validation: %s\n", is_valid ? "VALID" : "INVALID");
    
    // Test loading embedded model
    embedded_model_info_t model_info = c_load_embedded_model();
    printf("Loaded model info:\n");
    printf("  Data pointer: %p\n", (void*)model_info.data);
    printf("  Size: %zu bytes\n", model_info.size);
    printf("  Valid: %s\n", model_info.is_valid ? "YES" : "NO");
    
    // Test GGUF format validation
    int format_valid = c_validate_gguf_format(model_info.data, model_info.size);
    printf("GGUF format validation: %s\n", format_valid ? "VALID" : "INVALID");
    
    printf("Model embedding interface test completed successfully!\n");
    return 0;
}

File: src/test_model_embedding.zig
// test_model_embedding.zig - Simple test for model embedding
const std = @import("std");
const model_embedding = @import("model_embedding.zig");

pub fn main() !void {
    std.debug.print("Testing model embedding functionality...\n", .{});
    
    // Test getting embedded model
    const model_data = model_embedding.getEmbeddedModel();
    std.debug.print("Got embedded model of size: {} bytes\n", .{model_data.len});
    
    // Test model size
    const model_size = model_embedding.getModelSize();
    std.debug.print("Model size from function: {} bytes\n", .{model_size});
    
    // Verify sizes match
    if (model_data.len != model_size) {
        std.debug.print("ERROR: Size mismatch!\n", .{});
        return error.SizeMismatch;
    }
    
    // Test model loader
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();
    
    var loader = model_embedding.ModelLoader.init(allocator);
    const loaded_model = try loader.loadEmbeddedModel();
    
    std.debug.print("Loaded model of size: {} bytes\n", .{loaded_model.len});
    
    // Test validation
    const is_valid = try loader.validateModel(loaded_model);
    std.debug.print("Model validation: {}\n", .{is_valid});
    
    std.debug.print("Model embedding test completed successfully!\n", .{});
}